assignments/homework/chernyavskaya/hw5.py

```diff
@@ -0,0 +1,119 @@
+# imports
+import numpy as np
+from astropy import units as u
+from astropy.modeling import fitting, models
+from astropy.io import fits
+from astropy.coordinates import SkyCoord
+from photutils.datasets import load_star_image
+from astropy.stats import sigma_clip, SigmaClip, sigma_clipped_stats
+from photutils import CircularAperture, aperture_photometry, DAOStarFinder, Background2D, MedianBackground
+
+import matplotlib.pyplot as plt
+plt.rc('image', origin='lower')
+plt.rc('figure', figsize=(12, 8))
+
+#solutions
+# Challenge: Try fitting a Lorentzian and a Trapezoidal model to the same data. You can use tab-completion to find these models, or search in the documentation page. Overplot these models on the data along with the Gaussian fit.
+x1 = np.linspace(-5., 5., 200)
+y1 = 3 * np.exp(-0.5 * (x1 - 1.3)**2 / 0.8**2)
+y1 += np.random.normal(0., 0.2, x1.shape)
+yerr = np.random.uniform(0.15, 0.25, x1.shape)
+
+# fitter
+fitter = fitting.LevMarLSQFitter()
+# models
+g_init = models.Gaussian1D()
+l_init = models.Lorentz1D()
+t_init = models.Trapezoid1D()
+
+# fit
+g_fit = fitter(g_init, x1, y1)
+l_fit = fitter(l_init, x1, y1)
+t_fit = fitter(t_init, x1, y1)
+
+# errors
+g_fit_witherr = fitter(g_init, x1, y1, weights=1/yerr)
+l_fit_witherr = fitter(l_init, x1, y1, weights=1/yerr)
+t_fit_witherr = fitter(t_init, x1, y1, weights=1/yerr)
+
+# plotting
+plt.errorbar(x1, y1, yerr=yerr, fmt='.')
+xfine = np.linspace(-5, 5, 1000)
+plt.plot(xfine, g_init(xfine), lw=2, label = 'g_init', c='g')
+plt.plot(xfine, g_fit(xfine), lw=5, label = 'g_fit', c='b')
```

Comment:
This is an absolute nitpicking comment: the python style guide asks for spaces around operators and assignments, but not when indicating a keyword argument. around keyword argument (https://www.python.org/dev/peps/pep-0008/#other-recommendations)

Longer-term it's good to follow these guidelines, even though some of the recommendations don't seem to make much sense at first sight.  There are tools that check for it, you can set up a linter in your PyCharm, and/or editors, too. 

```suggestion
plt.plot(xfine, g_fit(xfine), lw=5, label='g_fit', c='b')
```